ClosedXML/Excel/XLWorkbook.cs

```diff
@@ -353,12 +353,24 @@ public IXLRange RangeFromFullAddress(String rangeAddress, out IXLWorksheet ws)
             if (!rangeAddress.Contains('!')) return null;
 
             var split = rangeAddress.Split('!');
-            var first = split[0];
-            var wsName = first.StartsWith("'") ? first.Substring(1, first.Length - 2) : first;
-            var localRange = split[1];
+            var wsName = split[0].UnescapeSheetName();
             if (TryGetWorksheet(wsName, out ws))
             {
-                return ws.Range(localRange);
+                return ws.Range(split[1]);
```

Comment:
I think we should check `split.Length` first. If `rangeAddress` is something like `Sheet1!`, we'd want an informative message in the exception.

Reply:
There is a safeguard already:
`if (!cellAddress.Contains('!')) return null;`

And if split `sheet1!` we still have an array of two elements (the second is an empty string. Added a few more test cases.